adam-core/src/main/scala/org/bdgenomics/adam/rdd/features/FeatureParser.scala

```diff
@@ -327,35 +338,44 @@ private[rdd] class BEDParser extends FeatureParser {
  * Specification:
  * http://www.genome.ucsc.edu/FAQ/FAQformat.html#format12
  */
-private[rdd] class NarrowPeakParser extends Serializable with Logging {
+private[rdd] class NarrowPeakParser extends FeatureParser {
 
-  def parse(line: String,
-            stringency: ValidationStringency): Seq[Feature] = {
-    val fields = line.split("\t")
-    def hasColumn(n: Int): Boolean = {
-      fields.length > n && fields(n) != "."
+  def isHeader(line: String): Boolean = {
+    line.startsWith("#") || line.startsWith("browser") || line.startsWith("track") || line.isEmpty
+  }
+
+  override def parse(line: String,
+                     stringency: ValidationStringency): Option[Feature] = {
+
+    // skip header lines
+    if (isHeader(line)) {
+      return None
     }
 
-    // skip empty or invalid lines
+    val fields = line.split("\t")
+
+    // check for invalid lines
     if (fields.length < 3) {
       if (stringency == ValidationStringency.STRICT) {
-        throw new IllegalArgumentException("Empty or invalid NarrowPeak line: %s".format(line))
+        throw new IllegalArgumentException("Invalid NarrowPeak line: %s".format(line))
       } else if (stringency == ValidationStringency.LENIENT) {
-        log.warn("Empty or invalid NarrowPeak line: {}", line)
+        log.warn("Invalid NarrowPeak line: {}", line)
       }
-      return Seq()
+      return None
+    }
+
+    def hasColumn(n: Int): Boolean = {
+      fields.length > n && fields(n) != "."
     }
-    val fb = Feature.newBuilder()
-    fb.setContigName(fields(0))
 
-    // Peak files are 0-based space-coordinates, so conversion to
-    // our coordinate space should mean that the values are unchanged.
-    fb.setStart(fields(1).toLong)
-    fb.setEnd(fields(2).toLong)
+    val f = Feature.newBuilder()
+      .setContigName(fields(0))
+      .setStart(fields(1).toLong) // NarrowPeak ranges are 0-based
```

Comment:
Will check to see what htsjdk does with NumberFormatExceptions and validation stringency.